Adds renewable cards

kbwatts opened the pull request:
Users really need to be able to bring the cards back, not only just to discard them permanently, and they should come back smoothly and animate-in just as nicely as they animate-out. This pull-request uses Polymer to provide a simple way for cards to smoothly animate back to the end of the list (in this example, using when the swipeable-card id="renewable").

Commit:
Adds renewable cards

Uses Polymer to provide a simple way for cards to smoothly animate back
to the end of the list.

cdata commented on demo.html:
```diff
@@ -1,53 +1,64 @@
     document.querySelector('#cards').addEventListener('swipeable-card-swipe-away', function(e) {
-      e.target.parentNode.removeChild(e.target);
+        e.target.parentNode.removeChild(e.target); // removes this card
+        if (e.target.id == "renewable") {
```
It might cool to express this behavior in the form of a Polymer element, rather than just a chunk of script. For example, a `renewable-card-list` or something similar.

kbwatts replied:
Nice idea - there may also be a more elegant way of doing this anyway (e.g. without using setTimeout), so it's probably worth optimising the code in addition to creating a separate Polymer element.

addyosmani replied:
I would +1 the idea of doing this via an element rather than in script. On the timeout, a 10s timeout for animations to complete seems rather arbitrary. Was there a particular reason this was approached this way?

kbwatts replied:
You're entirely right - the 10ms timeout is indeed arbitrary, chosen only as a small, non-zero number that still allowed the animation to proceed as expected. I arrived at 10 after a bit of testing with longer timeouts, but it could probably be 1, or done another way without timeouts altogether. I'll revisit as soon as I can.

Commit:
Script behaviour expressed via the Polymer element

script moved within the swipeable-cards element; ’removable’ and
‘renewable’ are now attributes; simplified demo page; timeout is 1ms.
May want to look at creating a renewable-card-list as mentioned by cdata

Commit:
Set removable to true by default

More similar to Google Now Cards behaviour

Commit:
Renewable card now returns to the deck it came from

rather than an arbitrarily named ‘#cards’ deck

kbwatts committed:
adds gitignore

kbwatts committed:
Functioning 1.0

TODO: disableSwipe attribute not working yet
